src/main/java/io/airlift/airline/TypeConverter.java

```diff
@@ -72,7 +73,10 @@ else if (Double.class.isAssignableFrom(type) || Double.TYPE.isAssignableFrom(typ
             Constructor<?> constructor = type.getConstructor(String.class);
             return constructor.newInstance(value);
         }
-        catch (Throwable ignored) {
+        catch (InvocationTargetException ex) {
+            throw new ParseOptionConversionException(name, value, type.getSimpleName(), ex.getTargetException());
+        }
+        catch (IllegalAccessException | IllegalArgumentException | InstantiationException | NoSuchMethodException | SecurityException ignored) {
```

Comment:
Is there a reason to have `IllegalArgumentException` and `SecurityException` on the list?

Reply:
Have used the IDE to replace catch the superclass with a multi-catch and than break the list up. But, indeed those two are irrelevant at that very point now.

Reply:
thanks for clarifying. The remaining ones you can conveniently catch with `ReflectiveOperationException` (without `|`s)

Reply:
I don't like this approach very much. I prefer a clear list of exceptions over catching the superclass. In this case, `InvocationTargetException` also extends the `ReflectiveOperationException`, which could be confusing. The implicit behavior would be correct, but it's cleaner to list them explicitly.

I will remove `IllegalArgumentException` and `SecurityException` from the list as they are unchecked anyway.